File: LeetCode/852(3).py
#Link: https://leetcode.com/problems/peak-index-in-a-mountain-array/

class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        mid = len(arr) // 2
        while True:
            if arr[mid-1] < arr[mid] and arr[mid] > arr[mid+1]:
                return mid
            if arr[mid] < arr[mid+1]:
                mid += 1
            else:
                mid -= 1

File: LeetCode/cpp/852(3).cpp
//Link: https://leetcode.com/problems/peak-index-in-a-mountain-array/

class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int mid = arr.size() / 2;
        while(true) {
            if(arr[mid-1] < arr[mid] && arr[mid] > arr[mid+1])
                return mid;
            if(arr[mid] > arr[mid-1])
                mid++;
            else
                mid--;
        }
    }
};
